### PR TITLE
[12.0][MIG] pos picking load

### PR DESCRIPTION
**Pull request by legalsylvain**
This PR migrates pos_picking_load from V10 to V12. (The last commit : https://github.com/OCA/pos/commit/229c5da412dbcba3eaf188f4dda899ca210e4709)

pos_picking_load allow to load in the point of sale, confirmed but non delivered out pickings.
load_picking_01_load_button

The list of picking is displayed in the point of sale.

load_picking_02_picking_list

When selected, it generates a pos order based on the stock moves , that can be paid.

load_picking_04_pos_order

See the readme part for a more complete description.

CC : @StefanRijnhart, @chienandalu 

Thanks for your review. 

#GRAPOCA

### CODE REVIEW

File: pos_picking_load/tests/test_pos_picking_load.py [hunk 1]
```diff
@@ -7,19 +6,22 @@
...
             'order_line': [(0, 0, {
                 'product_id': product.id,
                 'price_unit': 6,
                 'product_uom_qty': 2,
+                'qty_delivered_method': 'stock_move',
             })],
         })
         sale_order.action_confirm()
-        self.assertTrue(sale_order.picking_ids)
+        # self.assertTrue(sale_order.picking_ids)
```
Comment (kirca) [inline pos_picking_load/tests/test_pos_picking_load.py]: Why skip this check?

File: pos_picking_load/tests/test_pos_picking_load.py [hunk 2]
```diff
@@ -75,7 +79,9 @@ def test_pos_picking_load(self):
             [('origin_picking_id', 'in', sale_order.picking_ids.ids)])
         self.assertTrue(pos_order)
         self.assertEqual(pos_order.origin_picking_id.state, 'cancel')
-        self.assertIn(pos_order.picking_id, sale_order.picking_ids)
+        # self.assertIn(pos_order.picking_id, sale_order.picking_ids)
```
Comment (kirca) [inline pos_picking_load/tests/test_pos_picking_load.py]: Here too. I think if it is not valid anymore it is better to just remove it then commenting it out.

**Review by kirca — COMMENTED** [review]
Just added some comments, otherwise looks good. Quite nice module so thanks for initially contributing and maintaining it :).

**Comment by legalsylvain** [comment]
Hi @kirca. thanks for your review. I updated my code. Indeed, the commented code is obsolete. I removed it.
Did you tested it functionnaly on runbot ? 
regards

File: pos_picking_load/static/src/xml/pos_picking_load.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                        <i class="fa fa-arrow-circle-o-right"></i>
+                        Select
+                    </span>
+                    <span class="searchbox">
+                        <input placeholder="Search Picking" />
+                    </span>
+                </section>
+                <section class="full-content">
+                    <div class="window">
+                        <section class="subwindow collapsed">
```
Comment (kirca) [inline pos_picking_load/static/src/xml/pos_picking_load.xml:38]: This doesn't seem to be needed, it pushes the selection list down.
pos_picking_load

File: pos_picking_load/static/src/xml/pos_picking_load.xml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                    <span class="button back picking-button">
+                        <i class="fa fa-times"></i>
+                        Cancel
+                    </span>
+                    <span class="button validate picking-button highlight">
+                        <i class="fa fa-arrow-circle-o-right"></i>
+                        Select
+                    </span>
+                    <span class="searchbox">
+                        <input placeholder="Search Picking" />
```
Comment (kirca) [inline pos_picking_load/static/src/xml/pos_picking_load.xml:33]: The search doesn't seem to be handled in JS and it is not doing anything.

**Comment by kirca** [comment]
Hi @legalsylvain , test on runbot revealed some issues. Please check my comments.

File: pos_picking_load/static/src/xml/pos_picking_load.xml (hunk 3 above)
Comment (legalsylvain) [reply pos_picking_load/static/src/xml/pos_picking_load.xml:38]: Hi @kirca ! 

Well, I copy pasted windows present in Odoo. (partner selection).

On runbot, in my computer, it works like a charm.

pos_picking_load

So, I guess it's browser trouble don't you think ? 

for my Side, Mozilla Firefox for Ubuntu, 67.0.4 (64 bits)

do you face the same problem, when you select a customer ?

thanks !

File: pos_picking_load/static/src/xml/pos_picking_load.xml (hunk 4 above)
Comment (legalsylvain) [reply pos_picking_load/static/src/xml/pos_picking_load.xml:33]: Oh yes, my bad. Forgot to port this section.

File: pos_picking_load/static/src/xml/pos_picking_load.xml (hunk 3 above)
Comment (kirca) [reply pos_picking_load/static/src/xml/pos_picking_load.xml:38]: Indeed, issue is with Chrome. I did some digging and found commit where there is fix for it in the customer selection screen https://github.com/odoo/odoo/commit/4ff34a98bcefa96c908baae1646a5af8acd21913#diff-401e9bb2837c2ac0b281c1c8a9c195ceR440 . Just need to add "collapsed" class next to "subwindow-container".

Ubuntu 16.04
Chrome - Version 72.0.3626.119 (Official Build) (64-bit)
Firefox - 67.0.4 (64-bit)

**Comment by legalsylvain** [comment]
Hi @kirca  thanks a lot for your review, and your help regarding Chrome trouble.
- I fixed, the chrome bug, as you proposed ; 
- I restored search feature ; 
- I renamed min_date by scheduled_date for picking model ; 
- I added an extra very light module ``pos_picking_load_partner_name``. This module store the name of the partner on each picking, and change the search feature in pos_picking_load. the reason is that the domain [('partner_id', 'ilike', my_query)] is very slow on a huge database with millions of pickings. (~10 seconds) with that extra optional module, the search is performed in < 0.3 second.

regards.

Should be OK now.

**Comment by kirca** [comment]
Hi @legalsylvain , all changes look good. Also tested again on runbot and everything is working great. Nice job!

**Comment by rafaelbn** [comment]
Hi @kirca , is all is OK, aprrove the PR please. Thanks!

**Review by kirca — APPROVED** [review]
@rafaelbn all looks good to me

**Comment by legalsylvain** [comment]
@rafaelbn, @chienandalu : any update ?
regards.

**Review by chienandalu — APPROVED** [review]
This merge failed on tests. Maybe you can rebase (and squash some of the commits) and see if we can merge already. Otherwise good for me

**Comment by chienandalu** [comment]
travis not happy :/

**Comment by legalsylvain** [comment]
Hi all. After investigation, it seems that the error is present if ``pos_picking_delayed`` is installed. 
I made a PR for that purpose. (picking is not created, so procurement group test fails).
Could you review this trivial patch ? 

Thanks ! 
https://github.com/OCA/pos/pull/486

**Comment by chienandalu** [comment]
Done :)

File: pos_picking_load/models/sale_order.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    final_pos_order_id = fields.Many2one(
+        string='Final PoS Order', comodel_name='pos.order', readonly=True,
+        help="This Sale Order has beend replaced by this PoS Order")
```
Comment (gaikaz) [inline pos_picking_load/models/sale_order.py:13]: I can't seem to find, how this field is ever set. I suppose it was meant to be set in `_handle_orders_with_original_picking` method of `pos.order`?

**Review by gaikaz — COMMENTED** [review]
Very useful and great modules. :+1: 

I know this is already approved (though yet to merged for some issues with tests), but still:
In general **LGTM and functionally does what it says**, except for that one field on Sale Order.

Also, it would be nice to have it log a note (comment) in the canceled picking chatter, with a link to a new picking that replaces the "origin" one.
But maybe that's for me to implement as a custom addition :smile: Or at least not in this PR.

**Comment by gaikaz** [comment]
@legalsylvain Do you by any chance have a planned time to return to this PR? Maybe we just need to nudge someone a little to just merge it already?

Asking, because we kind of need this for v13. I could probably do a PR for migration to v13, but it would depend on this one.

**Comment by legalsylvain** [comment]
Hi @gaikaz 

This should be OK now.

/ocabot merge patch

Kind regards.

**Pull request merged**